webpack/scenes/ActivationKeys/Details/components/EditModal.js

```diff
@@ -0,0 +1,194 @@
+    handleModalToggle();
+  };
+
+  const handleNameInputChange = (value) => {
+    setNameValue(value);
+  };
+  const handleDescriptionInputChange = (value) => {
+    setDescriptionValue(value);
+  };
+
+  const onMinus = () => {
+    maxHostsValue(oldValue => (oldValue || 0) - 1);
+  };
+  const onChange = (event) => {
+    let newValue = (event.target.value === '' ? event.target.value : Math.round(+event.target.value));
+    if (newValue < 0) {
+      newValue = 0;
+    }
+    setMaxHostsValue(newValue);
+  };
+  const onPlus = () => {
+    setMaxHostsValue(oldValue => (oldValue || 0) + 1);
+  };
+
+  const handleCheckBox = () => {
+    setUnlimited(prevUnlimited => !prevUnlimited);
+  };
+
+  return (
+    <>
+      <Button ouiaId="ak-edit-button" variant="secondary" onClick={handleModalToggle}>
+        Edit
+      </Button>
+      <Modal
+        ouiaId="ak-edit-modal"
+        variant={ModalVariant.small}
+        title="Edit activation key"
+        description={`Select attributes for ${akDetails.name}`}
+        isOpen={isModalOpen}
+        onClose={handleClose}
+        actions={[
+          <Button ouiaId="edit-modal-save-button" key="create" variant="primary" form="modal-with-form-form" onClick={handleSave}>
+            Save
+          </Button>,
+          <Button ouiaId="cancel-button" key="cancel" variant="link" onClick={handleClose}>
+            Cancel
+          </Button>,
+        ]}
+      >
+        <Form isHorizontal>
+          <FormGroup
+            label="Name"
+          >
+            <TextInput
+              ouiaId="ak-name-input"
+              id="ak-name-input"
+              type="text"
+              value={nameValue}
+              onChange={handleNameInputChange}
+            />
+          </FormGroup>
+          <FormGroup
+            label="Host limit"
+          >
+            <Stack hasGutter>
+              <StackItem>
+                <NumberInput
+                  value={maxHostsValue}
+                  min={0}
+                  onMinus={onMinus}
+                  onChange={onChange}
+                  onPlus={onPlus}
+                  inputName="input"
+                  inputAriaLabel="number input"
+                  minusBtnAriaLabel="minus"
+                  plusBtnAriaLabel="plus"
+                  isDisabled={isUnlimited}
+                  allowEmptyInput
+                />
+              </StackItem>
+              <StackItem>
+                <Checkbox
+                  ouiaId="unlimited-checkbox"
+                  id="unlimited-checkbox"
+                  label="Unlimited"
+                  isChecked={isUnlimited}
+                  onChange={handleCheckBox}
+                />
+              </StackItem>
+            </Stack>
+          </FormGroup>
+          <FormGroup
+            label="Description"
+          >
+            <TextArea
+              id="ak-description"
+              type="text"
+              defaultValue={descriptionValue || 'Description empty'}
```

Comment:
'Description empty' could just be 'Description'. It should render disabled like a suggestion.
Also, these strings are easy to miss for translation. We likely have several of these in the code elsewhere..